### PR TITLE
Clean up scanlog

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
- Terminate throughput printing goroutine
- Fix style

### CODE REVIEW

File: scanner/scanner.go [hunk 1]
```diff
@@ -269,11 +269,48 @@ func humanTime(seconds int) string {
...
 
+func (s *Scanner) logThroughput(treeSize int64, stop <-chan bool) {
+	const wndSize = 15
+	wnd := make([]int64, wndSize)
+	wndTotal := int64(0)
+
+	ticker := time.NewTicker(time.Second)
+	for slot, slots, prevCnt := 0, 0, int64(0); ; slot = (slot + 1) % wndSize {
+		select {
+		case <-stop:
```
Comment (pavelkalinnikov) [inline scanner/scanner.go]: Perhaps a better approach would be to drop the `stop` channel and just close the `ticker` from the outside.

File: scanner/scanner.go [hunk 2]
```diff
@@ -358,15 +373,16 @@ func (s *Scanner) Scan(ctx context.Context, foundCert func(*ct.LogEntry), foundP
...
+
 	return nil
 }
 
-// NewScanner creates a new Scanner instance using client to talk to the log,
-// taking configuration options from opts.
-func NewScanner(client *client.LogClient, opts ScannerOptions) *Scanner {
+// New creates a Scanner instance using client to talk to the log, taking
+// configuration options from opts.
+func New(client *client.LogClient, opts ScannerOptions) *Scanner {
```
Comment (daviddrysdale) [inline scanner/scanner.go]: I suspect Travis is about to tell you that there are a couple of other users of this API that will need migrating...

File: scanner/scanner.go [hunk 3]
```diff
@@ -284,49 +321,27 @@ func (s *Scanner) Scan(ctx context.Context, foundCert func(*ct.LogEntry), foundP
...
-				certsMatched, (100.0*float64(certsMatched))/float64(certsProcessed),
-				throughput, remainingString))
-			i++
-		}
-	}()
 
 	var ranges list.List
-	for start := s.opts.StartIndex; start < int64(latestSth.TreeSize); {
-		end := min(start+int64(s.opts.BatchSize), int64(latestSth.TreeSize)) - 1
+	// TODO(pavelkalinnikov): Add EndIndex parameter.
```
Comment (daviddrysdale) [inline scanner/scanner.go]: See also: https://github.com/google/certificate-transparency-go/pull/172/commits/60abe5138a06f930659db8c8ad353c6027da18fd
(part of something else that's been pending for a while)

File: scanner/scanner.go [hunk 4]
```diff
@@ -269,11 +269,48 @@ func humanTime(seconds int) string {
 	return s
 }
 
+func (s *Scanner) logThroughput(treeSize int64, stop <-chan bool) {
+	const wndSize = 15
+	wnd := make([]int64, wndSize)
+	wndTotal := int64(0)
+
+	ticker := time.NewTicker(time.Second)
+	for slot, slots, prevCnt := 0, 0, int64(0); ; slot = (slot + 1) % wndSize {
```
Comment (daviddrysdale) [inline scanner/scanner.go]: `slot` and `slots` are hard to follow -- maybe `curSlot` and `filledSlots` ?

File: scanner/scanner.go (hunk 1 above)
Comment (pavelkalinnikov) [reply scanner/scanner.go]: Apparently `ticker.Stop()` doesn't close `ticker.C`, and there is no way to close that channel manually. The way to terminate here is indeed having an additional channel (see Example [here](https://golang.org/pkg/time/#NewTicker)).

File: scanner/scanner.go (hunk 3 above)
Comment (pavelkalinnikov) [reply scanner/scanner.go]: I think these two can be complementary. If `EndIndex == -1` (by default), scan as usual up to `sth.TreeSize`, otherwise scan up to `min(sth.TreeSize, EndIndex)`. In both cases your `ScanLog` method will return the end of the scanned interval.

Comment (pavelkalinnikov) [reply scanner/scanner.go]: See #199

File: scanner/scanner.go [hunk 5]
```diff
@@ -269,13 +269,48 @@ func humanTime(seconds int) string {
...
+func (s *Scanner) logThroughput(treeSize int64, stop <-chan bool) {
+	const wndSize = 15
+	wnd := make([]int64, wndSize)
+	wndTotal := int64(0)
+
+	ticker := time.NewTicker(time.Second)
+	for slot, filled, prevCnt := 0, 0, int64(0); ; slot = (slot + 1) % wndSize {
+		select {
+		case <-stop:
+			break
```
Comment (daviddrysdale) [inline scanner/scanner.go]: As Travis points out, this only breaks out of the `select` not the `for`

Comment (pavelkalinnikov) [reply scanner/scanner.go]: Yes, I just fixed that.
